core-bundle/contao/drivers/DC_Folder.php

```diff
@@ -3095,4 +3078,44 @@ public function initPicker(PickerInterface $picker)
 
 		return $attributes;
 	}
+
+	protected function updateFilemounts(array $filemounts)
+	{
+		$this->arrFilemounts = $this->eliminateNestedPaths(array_unique($filemounts));
+		$visibleRootTrails = array();
+
+		// Fetch visible root trails if enabled
+		if ($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['showRootTrails'] ?? false)
+		{
+			foreach ($this->arrFilemounts as $filemount)
+			{
+				$visibleRootTrails = array(...$visibleRootTrails, ...$this->getParentFilemounts($filemount));
+			}
+
+			// If there are no $visibleRootTrails (= no file mounts = everything should
+			// be visible), we have to fall back to all top level folders
+			if (array() === $visibleRootTrails)
+			{
+				$visibleRootTrails = array_map(fn (string $folder) => $this->strUploadPath . '/' . $folder, Folder::scan($this->strUploadPath));
```

Comment:
This is not correct because this will allow **everything** if the user does not have a file mount. Should we add an `->isAdmin` check here?

Reply:
I guess you mixed this up with the file mounts - but this is about root trails :)

Reply:
Ah, now I understand. Fixed in 16cf0c6cbc5c03a581ce3fa874c270addeb7b404. I used the same CS as in https://github.com/contao/contao/pull/6165 already.